Fix test regression: hasGarage missing from test assertions

## Problem

Commit cb4d583 (Game-defs parser diff mode, DLC registry, garage-only economics #118) added `hasGarage: boolean` to the City interface but did not update test assertions. Tests on main are now failing.

## Reproduction

```
npm run test
```

Result: 1 failed test in data.test.ts:
```
FAIL src/frontend/__tests__/data.test.ts > data.ts > loadAllData > loads observations-only and returns combined object
AssertionError: expected { id: 'berlin', name: 'Berlin', ...(2) } to deeply equal { id: 'berlin', name: 'Berlin', ...(1) }
- Expected: { country: "", id: "berlin", name: "Berlin" }
+ Received: { country: "", hasGarage: true, id: "berlin", name: "Berlin" }
```

## Fix Required

1. **data.test.ts line 120**: Add `hasGarage: true` to expected city object
2. **data.test.ts buildAllData helper** (lines 52-56): Update to include `hasGarage` in city construction
3. **data.test.ts buildLookups assertions** (lines 211-212): Update expected cities
4. **optimizer.test.ts lines 75-78**: Add `hasGarage: true` to all mock city objects
5. **Consider**: Create `tsconfig.test.json` or remove test file exclusion from tsconfig.json so type errors in tests are caught by `npm run lint`

Estimated effort: 15 minutes.

Source: analysis/qa-review.md

--- a/src/frontend/__tests__/data.test.ts
+++ b/src/frontend/__tests__/data.test.ts
@@ -53,6 +53,7 @@ function buildAllData(obs = sampleObservations) {
       id,
       name: titleCase(id),
       country: '',
+      hasGarage: true,
     })),
     companies: obs.companies.map((id) => ({
       id,
@@ -117,7 +118,7 @@ describe('data.ts', () => {
       expect(result.observations).toEqual(sampleObservations);
       expect(result.gameDefs).toBeNull();
       expect(result.cities).toHaveLength(2);
-      expect(result.cities[0]).toEqual({ id: 'berlin', name: 'Berlin', country: '' });
+      expect(result.cities[0]).toEqual({ id: 'berlin', name: 'Berlin', country: '', hasGarage: true });
       expect(result.companies).toHaveLength(2);
       expect(result.cargo).toHaveLength(4);
       expect(result.trailers).toHaveLength(3);
@@ -208,8 +209,8 @@ describe('data.ts', () => {
     it('creates citiesById map correctly', () => {
       const lookups = data.buildLookups(testData);
 
-      expect(lookups.citiesById.get('berlin')).toEqual({ id: 'berlin', name: 'Berlin', country: '' });
-      expect(lookups.citiesById.get('paris')).toEqual({ id: 'paris', name: 'Paris', country: '' });
+      expect(lookups.citiesById.get('berlin')).toEqual({ id: 'berlin', name: 'Berlin', country: '', hasGarage: true });
+      expect(lookups.citiesById.get('paris')).toEqual({ id: 'paris', name: 'Paris', country: '', hasGarage: true });
       expect(lookups.citiesById.get('unknown')).toBeUndefined();
     });
 

--- a/src/frontend/__tests__/optimizer.test.ts
+++ b/src/frontend/__tests__/optimizer.test.ts
@@ -73,9 +73,9 @@ describe('optimizer', () => {
     },
     observations: null,
     cities: [
-      { id: 'berlin', name: 'Berlin', country: 'germany' },
-      { id: 'paris', name: 'Paris', country: 'france' },
-      { id: 'empty_city', name: 'Empty City', country: '' },
+      { id: 'berlin', name: 'Berlin', country: 'germany', hasGarage: true },
+      { id: 'paris', name: 'Paris', country: 'france', hasGarage: true },
+      { id: 'empty_city', name: 'Empty City', country: '', hasGarage: true },
     ],
     companies: [
       { id: 'logistics_co', name: 'Logistics Co' },

